internal/http/services/owncloud/ocdav/ocdav_blackbox_test.go

```diff
@@ -120,13 +134,383 @@ var _ = Describe("ocdav", func() {
+				})
+
+			})
+
+			When("the parent does not exist", func() {
+				It("returns a conflict status", func() {
+
+					// MKCOL needs to check if the resource already exists to return the correct status
+					client.On("Stat", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.StatRequest) bool {
+						return req.Ref.Path == "/users/username/newfolder"
+					})).Return(&cs3storageprovider.StatResponse{
+						Status: status.NewAborted(ctx, errors.New("parent does not exist"), ""),
+					}, nil)
+
+					handler.Handler().ServeHTTP(rr, req)
+					Expect(rr).To(HaveHTTPStatus(http.StatusPreconditionFailed))
+					// TODO what message does oc10 return? "error: aborted:" is probably not it
+					Expect(rr).To(HaveHTTPBody("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<d:error xmlns:d=\"DAV\" xmlns:s=\"http://sabredav.org/ns\"><s:exception>Sabre\\DAV\\Exception\\PreconditionFailed</s:exception><s:message>error: aborted: </s:message></d:error>"), "Body must have a precondition failed sabredav exception")
+
+				})
+			})
+
+			When("the resource already exists", func() {
+				It("returns a method not allowed status", func() {
+
+					// MKCOL needs to check if the resource already exists to return the correct status
+					client.On("Stat", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.StatRequest) bool {
+						return req.Ref.Path == "/users/username/newfolder"
+					})).Return(&cs3storageprovider.StatResponse{
+						Status: status.NewOK(ctx),
+						Info:   &cs3storageprovider.ResourceInfo{},
+					}, nil)
+
+					handler.Handler().ServeHTTP(rr, req)
+					Expect(rr).To(HaveHTTPStatus(http.StatusMethodNotAllowed))
+					Expect(rr).To(HaveHTTPBody("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<d:error xmlns:d=\"DAV\" xmlns:s=\"http://sabredav.org/ns\"><s:exception>Sabre\\DAV\\Exception\\MethodNotAllowed</s:exception><s:message>The resource you tried to create already exists</s:message></d:error>"), "Body must have a method not allowed sabredav exception")
+
+				})
+			})
+		})
+
+		Describe("DELETE", func() {
+
+			var rr *httptest.ResponseRecorder
+			var req *http.Request
+			var err error
+
+			BeforeEach(func() {
+				// setup the request
+				rr = httptest.NewRecorder()
+				req, err = http.NewRequest("DELETE", basePath+"/existingfolder", strings.NewReader(""))
+				Expect(err).ToNot(HaveOccurred())
+				req = req.WithContext(ctx)
+
+			})
+
+			When("the gateway returns OK", func() {
+				It("returns a created status", func() {
+
+					client.On("Delete", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.DeleteRequest) bool {
+						return utils.ResourceEqual(req.Ref, &cs3storageprovider.Reference{
+							ResourceId: userspace.Root,
+							Path:       "./existingfolder",
+						})
+					})).Return(&cs3storageprovider.DeleteResponse{
+						Status: status.NewOK(ctx),
+					}, nil)
+
+					handler.Handler().ServeHTTP(rr, req)
+					Expect(rr).To(HaveHTTPStatus(http.StatusNoContent))
```

Comment:
The expectation doesn't seem to match the test description (created vs. no content)

Reply:
good catch